### PR TITLE
removed all usages of numpy.compat

### PR DESCRIPTION
**Pull request by ghost**
#### Reference issue
#13880 

#### What does this implement/fix?
This fix removes all usages of numpy.compat.

### CODE REVIEW

File: scipy/io/idl.py [hunk 1]
```diff
@@ -31,7 +31,7 @@
 
 import struct
 import numpy as np
-from numpy.compat import asstr
```
Comment (rgommers) [inline scipy/io/idl.py:34]: You can't just remove this, `asstr` is used in the rest of the code here so should be adapted.

Please try to run tests locally, that way you'll catch such things.

**Comment by rgommers** [comment]
@sanre3 please read our contributor guide (http://scipy.github.io/devdocs/dev/contributor/contributor_toc.html) and try to make this work locally, don't just push random commits. Your changes are very obviously incorrect.

**Comment by harishsdev** [comment]
@rgommers  can you please guide where to implement the functionality of asstr and asbytes

**Comment by tupui** [comment]
Seems like the user deleted his/her GitHub account...  As the PR does not solve the issue, we can close this in a few days if we don't have any news then.

**Comment by peterbell10** [comment]
IMO deleting their account is a clear indication they don't plan to continue so I'll close now.  If they reopen the account, they can always reopen the PR as well.

**Pull request closed without merging**